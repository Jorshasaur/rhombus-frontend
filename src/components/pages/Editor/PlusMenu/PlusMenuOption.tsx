import * as React from 'react'
import styles from './PlusMenuOption.module.css'

interface Props {
    onClickAction: () => void
    onMouseOver?: () => void
    icon: JSX.Element
    testId?: string
}

export default class PlusMenuOption extends React.Component<Props> {
    _onClick = (e: React.MouseEvent<HTMLElement>) => {
        e.preventDefault()
        this.props.onClickAction()
    }

    render() {
        return (
            <li
                onMouseOver={() => {
                    if (this.props.onMouseOver) {
                        this.props.onMouseOver()
                    }
                }}
                data-testid={this.props.testId}
                className={styles.plusMenuButton}
                onClick={this._onClick}>
                {this.props.icon}
                <span>{this.props.children}</span>
            </li>
        )
    }
}
