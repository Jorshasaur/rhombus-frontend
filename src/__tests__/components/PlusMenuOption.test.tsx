import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PlusMenuOption from '../../components/pages/Editor/PlusMenu/PlusMenuOption'

Enzyme.configure({ adapter: new Adapter() })

const defaultProps = {
    icon: <div className="test-icon" />,
    onClickAction: jest.fn(),
    onMouseOver: jest.fn()
}

describe('PlusMenuOption', () => {
    it('should be an instance of PlusMenuOption', () => {
        const wrapper = shallow(<PlusMenuOption {...defaultProps} />)
        const inst = wrapper.instance()
        expect(inst).toBeInstanceOf(PlusMenuOption)
    })
    it('should render PlusMenuOption', () => {
        const wrapper = shallow(<PlusMenuOption {...defaultProps} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should call click action click', () => {
        const wrapper = mount(<PlusMenuOption {...defaultProps} />)
        wrapper.simulate('click')
        expect(defaultProps.onClickAction).toHaveBeenCalled()
    })
    it('should call the mouseover event on mouse over', () => {
        const wrapper = mount(<PlusMenuOption {...defaultProps} />)
        wrapper.simulate('mouseover')
        expect(defaultProps.onMouseOver).toHaveBeenCalled()
    })
})
