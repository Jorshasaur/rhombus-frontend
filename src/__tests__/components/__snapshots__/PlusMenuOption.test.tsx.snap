// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusMenuOption should render PlusMenuOption 1`] = `
<li
  className="plusMenuButton"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="test-icon"
  />
  <span />
</li>
`;
